core/cache/cachetest/state.go

```diff
@@ -122,7 +122,8 @@ func MachineChange(c *gc.C, modelUUID string, machine *state.Machine) cache.Mach
 		SupportedContainersKnown: scKnown,
 		HardwareCharacteristics:  hwc,
 		CharmProfiles:            chProf,
-		HasVote:                  machine.HasVote(),
+		HasVote:                  true,
```

Comment:
Shouldn't this be removed?

Reply:
We still expose has/wants vote info to api callers via the MachineInfo struct (composed from a few sources of info). The cache entity reflects those values. Maybe we can remove from cache but need to verify first.

Reply:
This isn't used AFAIK, but correctness would have it determined the same way as it is in backing state for the multi-watcher. Something like:
```go
	wantsVote := false
	hasVote := false
	node, err := st.ControllerNode(m.Id)
	if err != nil && !errors.IsNotFound(err) {
		return errors.Trace(err)
	}
	wantsVote = err == nil && node.WantsVote()
	hasVote = err == nil && node.HasVote()
	...
```